### DISCUSSION TITLE
Live receive and send audio and video

### DISCUSSION THREAD

**Discussion by erfanh3h** [fastapi/fastapi#9028] [Questions]
Hello guys.
I wanted to receive and send audio and video file through websockets or any other ways in fastapi.(proccess like a chatroom that someone talking and others listening lively).
does anyone have any suggestions?

**Comment by Kludex** [comment]
Suggestion on what? Resources? What do you want to know?

**Comment by erfanh3h** [comment]
> Suggestion on what? Resources? What do you want to know?


Tnx for answer.
I googling for this and also see docs in docs but found nothing relating to this situation and webrtc in fastapi.
Resources or examples both are good.

**Comment by waynerv** [comment]
This post might help:
[Streaming video with FastAPI](https://stribny.name/blog/fastapi-video/)

**Comment by erfanh3h** [comment]
> This post might help:
> [Streaming video with FastAPI](https://stribny.name/blog/fastapi-video/)


Tnx for answer.
In this example streaming happen when file already exists .But in my case ,user is recording audio or video and in same time others are listening or watching his(or her) audio or video.

**Comment by ycd** [comment]
You can do pretty much anything you want with: WebRTC, see: https://github.com/aiortc/aiortc

**Comment by erfanh3h** [comment]
> You can do pretty much anything you want with: WebRTC, see: https://github.com/aiortc/aiortc


Tnx for answer.
I'm looking for a way in fastapi and not interest in using another package.😅

**Comment by haviduck** [comment]
> > You can do pretty much anything you want with: WebRTC, see: https://github.com/aiortc/aiortc
> 
> Tnx for answer.
> I'm looking for a way in fastapi and not interest in using another package.😅

fastapi is a web api framework and wont be able to solve your needs alone. 
have a look at vidgears. you can mount and mix alot of those and its ffmpeg integration supports audio.  webrtc is your best bet for video and audio if you wanna just get something up and running quickly. it wont scale very well but itll work. webrtc has little to do with fastapi other than perhaps using it as handshake.  

non-answer but why not check out bbb, jitsi and red 5 and build fastapi as a bridge towards their apis?

**Comment by answerquest** [comment]
@ycd  @haviduck any sample code out there to implement aiortc? I'll share the use case: Need to setup a streaming audio recorder on a ubuntu based server which receives audio from a client html / app and saves it.

**Comment by xRv9Fk0oQ6Da** [comment]
> @ycd @haviduck any sample code out there to implement aiortc? I'll share the use case: Need to setup a streaming audio recorder on a ubuntu based server which receives audio from a client html / app and saves it.

@answerquest, I'd love a link to your Repo or anyone's complete Repo which has this already setup. That'd be extremely helpful.

> You can do pretty much anything you want with: WebRTC, see: https://github.com/aiortc/aiortc

Also, here in 2022 WebRTC would be perfect; but isn't it basically deprecated due to Security Vulnerabilities?
